Runbooks/startstop/1.1/AutoStop_VM_Child_ARM.ps1

```diff
@@ -71,39 +71,37 @@ if ($WebHookData) {
             $Context
 
             Write-Output 'Successfully logged into Azure subscription using Az cmdlets...'
-            $vmTags = Get-AzVM -ResourceGroupName $context.resourceGroupName -Name $context.resourceName | Select-Object Tags
 
-            if ($vmTags.Tags[$ExcludedTagName] -eq $ExcludedTagValue) {
-                Write-Output "Virtual Machine $($context.resourceName) excluded by tag."
-                $RetryFlag = $false
-            } elseif ($context.resourceType -eq 'Microsoft.Compute/virtualMachines') {
+            if ($context.resourceType -eq 'Microsoft.Compute/virtualMachines') {
                 Write-Output "~$($context.resourceName)"
+                $vmTags = Get-AzVM -ResourceGroupName $context.resourceGroupName -Name $context.resourceName | Select-Object Tags
 
-                Write-Output "Stopping Virtual Machine : $($context.resourceName)"
-
-                $Status = Stop-AzVM -Name $context.resourceName -ResourceGroupName $context.resourceGroupName -Force
+                if ($vmTags.Tags[$IncludedTagName] -eq $IncludedTagValue) {
+                    Write-Output "Virtual Machine $($context.resourceName) included by tag."
+                    Write-Output "Stopping Virtual Machine : $($context.resourceName)"
+                    $Status = Stop-AzVM -Name $context.resourceName -ResourceGroupName $context.resourceGroupName -Force
+                } else {
+                    Write-Output "Virtual Machine $($context.resourceName) not included by tag. Skipping."
+                    $Status = 'NotIncluded'
+                }
 
                 if ($null -eq $Status) {
                     Write-Output "Error occurred while stopping the Virtual Machine $($context.resourceName) hence retrying..."
-
                     if ($Attempt -gt $RetryCount) {
                         Write-Output "Reached the max $RetryCount retry attempts so please resubmit the job later..."
-
                         $RetryFlag = $false
                     } else {
                         Write-Output "[$Attempt/$RetryCount] Retrying in $TimeoutInSecs seconds..."
-
                         Start-Sleep -Seconds $TimeoutInSecs
-
                         $Attempt = $Attempt + 1
-
                         $RetryFlag = $true
                     }
                 } else {
-                    Write-Output "Successfully stopped the Virtual Machine : $($context.resourceName)"
-
-                    Write-Output "$context.resourceName"
-
+                    if ($Status -eq 'NotIncluded') {
```

Comment:
This should be -ne

Reply:
Though you are correct, I instead chose to switch around the content of the if to make it fit with the other scripts. Thank you for noticing!

Reply:
```pwsh
if ($Status -eq 'NotIncluded') {
    Write-Output "Virtual Machine $($VMName) not stopped because it was not included by tag."
} else {
    Write-Output "Successfully stopped the Virtual Machine : $($context.resourceName)"
}
```

Reply:
Updated again, as the variable name for VM was wrong,